pkg/image/api/v1/types.go

```diff
@@ -144,3 +182,44 @@ type DockerImageReference struct {
 	Tag       string
 	ID        string
 }
+
+type ImageStreamImport struct {
+	unversioned.TypeMeta `json:",inline"`
+	kapi.ObjectMeta      `json:"metadata,omitempty" description:"metadata about the image stream, name is required"`
+
+	Repository *RepositoryImport `json:"repository,omitempty" description:"if specified, import a single Docker repository's tags to this image stream"`
+	Images     []ImageImport     `json:"images,omitempty" description:"a list of images to import into this image stream"`
+}
+
+type RepositoryImport struct {
+	From kapi.ObjectReference `json:"from" description:"the source for the image repository to import; only kind DockerImage and a name of a Docker image repository is allowed"`
+	To   bool                 `json:"to,omitempty" description:"if true, the located images should be added to the image stream with the corresponding tags"`
+
+	ImportPolicy    TagImportPolicy `json:"importPolicy,omitempty" description:"policy controlling how the image is imported"`
+	IncludeManifest bool            `json:"includeManifest,omitempty" description:"if true, return the manifest for each image in the response"`
+
+	RepositoryImportStatus `json:",inline" description:"status of the attempt to import images"`
```

Comment:
> The specification is a complete description of the desired state,
including configuration settings provided by the user, [default
values](#defaulting) expanded by the system, and properties initialized or
otherwise changed after creation by other ecosystem components (e.g.,
schedulers, auto-scalers), and is persisted in stable storage with the API
object. If the specification is deleted, the object will be purged from the
system. The status summarizes the current state of the object in the
system, and is usually persisted with the object by an automated processes
but may be generated on the fly. At some cost and perhaps some temporary
degradation in behavior, the status could be reconstructed by observation
if it were lost.

> All objects that represent a physical resource whose state may vary from
the user's desired intent SHOULD have a "spec" and a "status".  Objects
whose state cannot vary from the user's desired intent MAY have only
"spec", and MAY rename "spec" to a more appropriate name.


On Mon, Jan 11, 2016 at 1:31 PM, David Eads <notifications@github.com>
wrote:

> In pkg/image/api/v1/types.go
> <https://github.com/openshift/origin/pull/6288#discussion_r49358287>:
>
> > +type ImageStreamImport struct {
> > +	unversioned.TypeMeta `json:",inline"`
> > +	kapi.ObjectMeta      `json:"metadata,omitempty" description:"metadata about the image stream, name is required"`
> > +
> > +	Repository *RepositoryImport `json:"repository,omitempty" description:"if specified, import a single Docker repository's tags to this image stream"`
> > +	Images     []ImageImport     `json:"images,omitempty" description:"a list of images to import into this image stream"`
> > +}
> > +
> > +type RepositoryImport struct {
> > +	From kapi.ObjectReference `json:"from" description:"the source for the image repository to import; only kind DockerImage and a name of a Docker image repository is allowed"`
> > +	To   bool                 `json:"to,omitempty" description:"if true, the located images should be added to the image stream with the corresponding tags"`
> > +
> > +	ImportPolicy    TagImportPolicy `json:"importPolicy,omitempty" description:"policy controlling how the image is imported"`
> > +	IncludeManifest bool            `json:"includeManifest,omitempty" description:"if true, return the manifest for each image in the response"`
> > +
> > +	RepositoryImportStatus `json:",inline" description:"status of the attempt to import images"`
>
> I don't think there's a strong benefit from the outcome to an end user to
> have spec /status (and this is not an object that lives on the cluster).
>
> It's an expectation of API objects. "I set the spec, the server gives me
> status". It's the same argument for having an SAR with spec and status.
> It's a consistent view of actor responsibilities
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6288/files#r49358287>.
>